bin/test_runner.rb

```diff
@@ -129,12 +131,17 @@ def wait
   private
 
   def cloud_name(name)
+    # The rocket monkey naming convention goes like this:
+    # prefix_cloud_region_testname...
+    # What this test calls cloud name is actually 'cloud_region'
+    # So split on _ and get those two, for example.
+    # rl10lin_Google_Silicon_monitoring...
+    # yields 'Google_Silicon' as cloud name.
     name.split('_')[1..2].join('_')
   end
 
   def total_deps_up(filter)
-    deployments = @rsclient.deployments.index(filter: ["name==#{filter}"])
-    deployments.length
+    deployments = @rsclient.deployments.index(filter: ["name==#{filter}"]).length
   end
 
   def build_jenkins_job(job_name, wait_seconds)
```

Comment:
I suggested that `(-(wait.abs)..0).each do |i|` is used so future use of this code can use a non-negative number for wait seconds AND that current calls to `build_jenkins_job` with negative numbers will continue to work.  The reply was that this will all be reworked so adding the change here will be irrelevant .  Just want to keep as a reminder that if whole numbers are passed in, that they are non-negative.